### ISSUE TITLE
@MaybeShewill-CV Sorry， I have read chapter B several times，but I still unknown where use the “f” in matrix H in curve fitting, Can you explain it in more detail, thank you very much .

### ISSUE THREAD

**Issue by fyqqyf** [MaybeShewill-CV/lanenet-lane-detection#284]
@MaybeShewill-CV Sorry， I have read chapter B several times，but I still unknown where use the “f” in matrix H in curve fitting, Can you explain it in more detail, thank you very much .

In you paper you wrote:
Given a lane pixel pi = [xi; yi; 1]T belong to  P,
the transformed pixel p'i = [x'i; y'i; 1]T belong to P' is equal to Hpi=[axi+byi+c, dyi+e, fyi+1],
why p'i third element is one but not fyi+1? 

And in your loss funtion I still unkonw where use the pts_projects's third element?
`gt_pts = tf.transpose(gt_pts)
        pts_projects = tf.matmul(H, gt_pts)

        # 求解最小二乘二阶多项式拟合参数矩阵
        Y = tf.transpose(pts_projects[1, :])
        X = tf.transpose(pts_projects[0, :])
        Y_One = tf.add(tf.subtract(Y, Y), tf.constant(1.0, tf.float32))
        Y_stack = tf.stack([tf.pow(Y, 3), tf.pow(Y, 2), Y, Y_One], axis=1)
        w = tf.matmul(tf.matmul(tf.matrix_inverse(tf.matmul(tf.transpose(Y_stack), Y_stack)),
                                tf.transpose(Y_stack)),
                      tf.expand_dims(X, -1))
        # 利用二阶多项式参数求解拟合位置并反算到原始投影空间计算损失
        x_preds = tf.matmul(Y_stack, w)
        preds = tf.transpose(tf.stack([tf.squeeze(x_preds, -1), Y, Y_One], axis=1))
        x_transformation_back = tf.matmul(tf.matrix_inverse(H), preds)

        loss = tf.reduce_mean(tf.pow(gt_pts[0, :] - x_transformation_back[0, :], 2))`
x_transformation_back = tf.matmul(tf.matrix_inverse(H), preds)，this inverse operation will use f, but can we set f as zero?

_Originally posted by @andongchen in https://github.com/MaybeShewill-CV/lanenet-lane-detection/issues/216#issuecomment-492045188_

**Comment by fyqqyf** [comment]
Well, I also don't know why there is a 'f', I don't think it makes things different.😂

**Comment by MaybeShewill-CV** [comment]
@fyqqyf hnet was not supplied here:)

**Comment by fyqqyf** [comment]
Thanks！I just think that maybe you know that cause you have already read this paper and did a perfect job! Anyway, thanks a lot! =)
